hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,881 @@
+                        component_sub.centre.value = line_energy
+                        component_sub.name = xray_sub
+                        component_sub.fwhm = line_FWHM
+                        component_sub.centre.free = False
+                        component_sub.sigma.free = False
+                        component_sub.A.twin_function = _get_weight(
+                            element, li)
+                        component_sub.A.twin_inverse_function = _get_iweight(
+                            element, li)
+                        component_sub.A.twin = component.A
+                        self.append(component_sub)
+            self.fetch_stored_values()
+
+    @property
+    def _active_background_components(self):
+        return [bc for bc in self.background_components
+                if bc.coefficients.free]
+
+    def add_polynomial_background(self, order=6):
+        """
+        Add a polynomial background.
+
+        the background is added to self.background_components
+
+        Parameters
+        ----------
+        order: int
+            The order of the polynomial
+        """
+        background = create_component.Polynomial(order=order)
+        background.name = 'background_order_' + str(order)
+        background.isbackground = True
+        self.append(background)
+        self.background_components.append(background)
+
+    def free_background(self):
+        """
+        Free the yscale of the background components.
+        """
+        for component in self.background_components:
+            component.coefficients.free = True
+
+    def fix_background(self):
+        """
+        Fix the background components.
+        """
+        for component in self._active_background_components:
+            component.coefficients.free = False
+
+    def enable_xray_lines(self):
+        """Enable the X-ray lines components.
+
+        """
+        for component in self.xray_lines:
+            component.active = True
+
+    def disable_xray_lines(self):
+        """Disable the X-ray lines components.
+
+        """
+        for component in self._active_xray_lines:
+            component.active = False
+
+    def fit_background(self,
+                       start_energy=None,
+                       end_energy=None,
+                       windows_sigma=[4., 3.],
+                       kind='single',
+                       **kwargs):
+        """
+        Fit the background in the energy range containing no X-ray line.
+
+        After the fit, the background is fixed.
+
+        Parameters
+        ----------
+        start_energy : {float, None}
+            If float, limit the range of energies from the left to the
+            given value.
+        end_energy : {float, None}
+            If float, limit the range of energies from the right to the
+            given value.
+        windows_sigma: list of two float
+            The upper and lower bounds around each X-ray lines to define
```

Comment:
Strictly, I guess this should be "lower and upper", as that is how it is used, and since that is the order that makes the most sense.